multiversx_sdk_cli/cli_contracts.py

```diff
@@ -358,45 +372,57 @@ def deploy(args: Any):
         payable_by_sc=args.metadata_payable_by_sc,
         gas_limit=int(args.gas_limit),
         value=int(args.value),
-        nonce=int(args.nonce),
+        nonce=nonce,
         version=int(args.version),
         options=int(args.options),
-        guardian=args.guardian,
+        guardian=guardian_address,
+        relayer=relayer_address,
     )
-    tx = _sign_guarded_tx(args, tx)
+    tx = _sign_guarded_tx(guardian, args, tx)
+    _sign_tx_by_relayer(relayer, tx)
 
     address_computer = AddressComputer(NUMBER_OF_SHARDS)
-    contract_address = address_computer.compute_contract_address(deployer=sender.address, deployment_nonce=args.nonce)
+    contract_address = address_computer.compute_contract_address(deployer=sender.address, deployment_nonce=tx.nonce)
 
     logger.info("Contract address: %s", contract_address.to_bech32())
     utils.log_explorer_contract_address(args.chain, contract_address.to_bech32())
 
     _send_or_simulate(tx, contract_address, args)
 
 
-def _sign_guarded_tx(args: Any, tx: Transaction) -> Transaction:
-    try:
-        guardian_account = cli_shared.prepare_guardian_account(args)
-    except NoWalletProvided:
-        guardian_account = None
-
-    if guardian_account:
-        tx.guardian_signature = bytes.fromhex(guardian_account.sign_transaction(tx))
-    elif args.guardian:
-        tx = cosign_transaction(tx, args.guardian_service_url, args.guardian_2fa_code)  # type: ignore
+def _sign_guarded_tx(guardian: Union[IAccount, None], args: Any, tx: Transaction) -> Transaction:
+    if guardian:
+        tx.guardian_signature = bytes.fromhex(guardian.sign_transaction(tx))
```

Comment:
In Python, I remember `sign_transaction()` returns `bytes`, not hex-string:

https://github.com/multiversx/mx-sdk-specs/blob/main/accounts/account.md

Is `bytes.fromhex()` correct here?

Reply:
here `Account.sign_transaction()` returns a string, so yes, `bytes.from_hex()` is correct.